src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,75 @@
+package duke;
+
+import duke.commands.Command;
+import duke.commands.CommandResult;
+
+import duke.data.TaskList;
+
+import duke.exceptions.DukeException;
+
+import duke.parser.Parser;
+
+import duke.storage.Storage;
+
+import duke.ui.Ui;
+
+/**
+ * Implements the Duke chat bot.
+ * @author Lim Yong Shen, Kevin
+ */
+public class Duke {
+
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui;
+
+    /**
+     * Constructs a new Duke chat bot with the specified data file name.
+     * @param fileName The specified data file name.
+     */
+    private Duke(String fileName) {
+        ui = new Ui();
+        storage = new Storage(fileName);
+        try {
+            tasks = storage.load();
+        } catch(DukeException e) {
+            ui.showLoadingError();
+            tasks = new TaskList();
+        }
+    }
+
+    /**
+     * Runs the Duke chat bot until the exit command is given.
+     */
+    private void run() {
+        ui.showWelcome();
+        boolean isExit = false;
+        while (!isExit) {
+            try {
+                String input = ui.readInput();
+                ui.showHorizontalBorder();
+                Command command = Parser.parse(input);
+                command.setTaskListToExecuteOn(tasks);
+                CommandResult commandResult = command.execute();
+                ui.showMessage(commandResult.getMessage());
+                isExit = command.isExit();
+                storage.save(tasks);
+            } catch (DukeException e) {
+                ui.showMessage(e.getMessage());
+            } finally {
+                ui.showHorizontalBorder();
+                ui.showEmptyLine();
+            }
+        }
+        ui.close();
+    }
+
+    /**
+     * Executes the program.
+     * @param args Command line arguments (unused).
+     */
+    public static void main(String[] args) {
+        new Duke("/Users/lyskevin/Desktop/Y2 Sem 1/CS2103T/duke/src/main/java/duke/data/tasks.txt").run();
+    }
```

Comment:
I believe truncating the path name starting from the root would work as well. eg "duke/data/tasks.txt". Anyone who clones your repo won't be able to get the storage function to work with this path!

Reply:
Hi, I did do this prior to this commit but then I think I accidentally messed with the IntelliJ file structure and multiple paths didn't work for me. I left this here in order to finish the level but I've fixed it and it's back to a relative path now :)